src/main/java/nl/nn/testtool/Checkpoint.java

```diff
@@ -177,6 +177,31 @@ public static String getTypeAsString(int type) {
 		return typeAsString;
 	}
 
+	public void setTypeAsString(String type) {
+		if (type == null)
+			return;
+
+		if (type.equalsIgnoreCase("None")) {
+			this.type = TYPE_NONE;
+		}else if (type.equalsIgnoreCase("Startpoint")) {
+			this.type = TYPE_STARTPOINT;
+		}else if (type.equalsIgnoreCase("Endpoint")) {
+			this.type = TYPE_ENDPOINT;
+		}else if (type.equalsIgnoreCase("Abortpoint")) {
+			this.type = TYPE_ABORTPOINT;
+		}else if (type.equalsIgnoreCase("Inputpoint")) {
+			this.type = TYPE_INPUTPOINT;
+		}else if (type.equalsIgnoreCase("Outputpoint")) {
+			this.type = TYPE_OUTPUTPOINT;
+		}else if (type.equalsIgnoreCase("Infopoint")) {
+			this.type = TYPE_INFOPOINT;
+		}else if (type.equalsIgnoreCase("ThreadStartpoint")) {
+			this.type = TYPE_THREADSTARTPOINT;
+		}else if (type.equalsIgnoreCase("ThreadEndpoint")) {
+			this.type = TYPE_THREADENDPOINT;
+		}

```

Comment:
I think this could be written in less code and in a more readable way by using an Enum instead of integers. Once it's an enum you can also cleanup the code *(only a bit, no need to refactor everything at once)*
`getTypeAsString(int type) { return this.type.getName(); } `

Reply:
I didn't want to create new structures on old piece of code. Didn't want to mess with Ladybug code :D